### DISCUSSION TITLE
辞書作ってよ🥴

### DISCUSSION THREAD

**Discussion by ikotome** [WSOFT-Project/alicescript#23] [Ideas]
dict = dict()
dict = { ["", ""] , ["", ""] }
みたいなのほしーなーーーーー

**Comment by taiseiue** [comment]
JSONっぽく `{ key: value }`ってかけると良さそう
```cs
{
    "one": 1,
    "two": 2
}
```

**Comment by taiseiue** [comment]
配列式の延長で、

```cs
var dict = [
    "one" : 1,
    "two" : 2
]
```
ってかくのが実装的には丸そう。
配列式にKeyValue入れてるのと変わらんし

**Reply by taiseiue** [reply]
これそもそも
`key: value`をKeyValuePairを作る記法ってことにして、要素全てがKeyValuePairな配列はディクショナリにすればええんやないか。
もし右みてarray型が欲しそうならKeyValuePairな配列のままにしておくと良さそう。

**Reply by taiseiue** [reply]
ただそれやと、範囲指定子との兼ね合いが厳しそうな気もする。
もっかい範囲指定子の構文を`..`にできひんか検討してもいい気がする

**Reply by ikotome** [reply]
structってなんやけ
